### PR TITLE
Use dedicated logger for worker output logs (resolves #1136)

### LINKED ISSUE #1136: Toil logs kinda suck
Debugging Toil pipelines is harder than it needs to be because the stacktrace returned by Toil Worker Output Log has each line prepended by unnecessary log information.

### PR DESCRIPTION
**Pull request by cket**
Resolves #1136

### CODE REVIEW

**Comment by cket** [comment]
@jvivian can you test this out when you get a chance?

File: src/toil/leader.py [hunk 1]
```diff
@@ -34,6 +34,15 @@
 
 logger = logging.getLogger( __name__ )
 
+# create new logger for worker output logs so we can format them appropriately
+outputLogger = logging.getLogger('workerOutput')
```
Comment (hannes-ucsc) [inline src/toil/leader.py]: I think this should be `__name__ + ".workerOutput" such that it is a child of the "toil" logger.

File: src/toil/leader.py [hunk 2]
```diff
@@ -34,6 +34,15 @@
...
+
+logger_handler = logging.StreamHandler()
```
Comment (hannes-ucsc) [inline src/toil/leader.py]: Camel case.

**Comment by hannes-ucsc** [comment]
Generally +1. But see comments.

**Comment by hannes-ucsc** [comment]
Remember to add a ticket reference into the description or Waffle won't stack PR and issue.

**Comment by cket** [comment]
Don't you just have to reference it in the title? I think they were stacked before

**Comment by hannes-ucsc** [comment]
It wasn't stacked before and after adding the reference, it was.

**Comment by hannes-ucsc** [comment]
But this suggests that a reference in the title should be enough:

https://github.com/waffleio/waffle.io/wiki/FAQs#prs-closing-keywords

**Comment by cket** [comment]
Yeah, #1119 is stacked without a reference. Maybe waffle just needed to be refreshed

**Comment by hannes-ucsc** [comment]
Ahh, but a reference in the title doesn't create a backlink from the issue to the PR.

**Comment by cket** [comment]
Jenkins, test this please

**Comment by cket** [comment]
The test is failing because it redirects logs by adding a handler to the root logger. It then parses the files looking for worker output, which aren't there because they are logged by the new outputLogger and specifically not propagated to the root logger.

The solution would be to propagate the logs back to the root logger, but that would also cause all the outputLogger's messages to be double logged.

**Comment by cket** [comment]
closing, continued in #1177

**Pull request closed without merging**